src/main/scala/scalaz/actors/Actors.scala

```diff
@@ -2,59 +2,45 @@ package scalaz.actors
 
 import scalaz.zio.{ IO, Promise, Queue, Ref, Schedule }
 
-object Actors {
+trait Actor[+E, -F[+ _]] {
+  def ![A](fa: F[A]): IO[E, A]
 
-  trait Actor[+E, -F[+ _]] {
-    def ![A](fa: F[A]): IO[E, A]
-    def stop(): IO[Nothing, Unit]
-  }
-
-  trait MessageHandler[S, +E, -F[+ _]] {
-    def receive[A](state: S, msg: F[A]): IO[E, (S, A)]
-  }
-
-  trait Supervisor[E] {
-    def supervise[A](io: IO[E, A], error: E): IO[E, A]
-  }
-
-  object Supervisor {
-
-    def none: Supervisor[Nothing] = retry(Schedule.never)
+  val stop: IO[Nothing, Unit]
+}
 
-    def retry[E](policy: Schedule[E, _]): Supervisor[E] =
-      new Supervisor[E] {
+object Actor {
+  val DefaultActorMailboxSize = ActorConfig.config.orThrow().mailboxSize
```

Comment:
While it makes perfect sense to have configurable mailbox size, I don't think it is a good idea to handle the configuration inside the library. IMHO, loading the configuration and applying it should be done in client code.